elf2uf2/elf2uf2.cpp

```diff
@@ -231,80 +333,100 @@ int bin2uf2(std::shared_ptr<std::iostream> in, std::shared_ptr<std::iostream> ou
     return pages2uf2(pages, in, out, family_id, abs_block_loc);
 }
 
+/**
+ * @brief Converts an ELF file to UF2 format.
+ * 
+ * @param in Input stream for the ELF file.
+ * @param out Output stream for UF2 data.
+ * @param family_id Family ID for UF2.
+ * @param package_addr Optional packaging address.
+ * @param abs_block_loc Optional absolute block location.
+ * @return 0 on success, error code otherwise.
+ */
 int elf2uf2(std::shared_ptr<std::iostream> in, std::shared_ptr<std::iostream> out, uint32_t family_id, uint32_t package_addr, uint32_t abs_block_loc) {
     elf_file elf;
     std::map<uint32_t, std::vector<page_fragment>> pages;
 
+    // Read the ELF file
     int rc = elf.read_file(in);
     bool ram_style = false;
     address_ranges valid_ranges = {};
     address_ranges flash_range; address_ranges ram_range;
+
+    // Determine the valid address ranges based on the family ID
     if (family_id == RP2040_FAMILY_ID) {
         flash_range = rp2040_address_ranges_flash;
         ram_range = rp2040_address_ranges_ram;
     } else {
         flash_range = rp2350_address_ranges_flash;
         ram_range = rp2350_address_ranges_ram;
     }
+
     if (!rc) {
         rc = rp_determine_binary_type(elf.header(), elf.segments(), flash_range, ram_range, &ram_style);
         if (!rc) {
             if (verbose) {
-                if (ram_style) {
-                    printf("Detected RAM binary\n");
-                } else {
-                    printf("Detected FLASH binary\n");
-                }
+                printf("Detected %s binary\n", ram_style ? "RAM" : "FLASH");
             }
             valid_ranges = ram_style ? ram_range : flash_range;
             rc = check_elf32_ph_entries(elf.segments(), valid_ranges, pages);
         }
     }
+
     if (rc) return rc;
+
     if (pages.empty()) {
         fail(ERROR_INCOMPATIBLE, "The input file has no memory pages");
     }
-    // No Thumb bit on RISC-V
+
+    // Determine the Thumb bit (for ARM architecture)
     elf32_header eh = elf.header();
-    uint32_t thumb_bit = eh.common.machine == EM_ARM ? 0x1u : 0x0u;
+    uint32_t thumb_bit = (eh.common.machine == EM_ARM) ? 0x1u : 0x0u;
+
     if (ram_style) {
         uint32_t expected_ep_main_ram = UINT32_MAX;
         uint32_t expected_ep_xip_sram = UINT32_MAX;
+
+        // Determine expected entry points
         for(auto& page_entry : pages) {
-            if ( ((page_entry.first >= SRAM_START) && (page_entry.first < ram_range[0].to)) && (page_entry.first < expected_ep_main_ram) ) {
+            if (((page_entry.first >= SRAM_START) && (page_entry.first < ram_range[0].to)) && (page_entry.first < expected_ep_main_ram)) {
```

Comment:
These lines could instead have the second set of brackets on the left removed as it's just a 3-way AND, to prevent the `(((` which is hard to read
```suggestion
            if ((page_entry.first >= SRAM_START) && (page_entry.first < ram_range[0].to) && (page_entry.first < expected_ep_main_ram)) {
```